app/views/prison/visits/_cancel_visit.html.erb

```diff
@@ -1,24 +1,22 @@
-<%= form_tag cancel_prison_visit_path(@visit, locale: 'en'), class: 'form' do -%>
+<%= form_for [:prison, @visit, (@visit.cancellation || @visit.build_cancellation).decorate], class: 'form' do |f| -%>
```

Comment:
Can the logic for the cancellation object go somewhere else? Perhaps the visit decorator?
 Doesn't look like the right place to have this logic...